### PR TITLE
add reversed() builtin function, as well as compatible __reversed__ methods on list/xrange.

### PR DESCRIPTION
**Pull request by toshok**

### CODE REVIEW

File: src/runtime/objmodel.cpp [hunk 1]
```diff
@@ -3410,12 +3411,26 @@ Box* getiter(Box* o) {
...
+        return r;
+
+    if (!typeLookup(o->cls, getitem_str, NULL)) {
+        raiseExcHelper(TypeError, "'%s' object is not iterable", getTypeName(o)->c_str());
+    }
+    int64_t len = unboxedLen(o); // this will throw an exception if __len__ isn't there
+
+    return new (seqreviter_cls) BoxedSeqIter(o, len - 1);
+}
+
```
Comment (kmod) [inline src/runtime/objmodel.cpp]: Tiny preference, but I would put this in the builtins.cpp file if nothing else will be calling this other than the reversed() function.

File: src/runtime/iterobject.cpp [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #include "capi/types.h"
 #include "core/common.h"
 #include "core/options.h"
+
```
Comment (kmod) [inline src/runtime/iterobject.cpp]: Our lint script doesn't like this newline...

**Comment by toshok** [comment]
Oops, should be fixed now.  I had run `make check_dbg`/`make check_release` individually but didn't run `make check`

**Comment by kmod** [comment]
Cool!

**Pull request merged**